content/sensu-go/5.17/reference/secret-providers.md

```diff
@@ -0,0 +1,297 @@
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"metadata": {
+  "name": "vault"
+}
+{{< /highlight >}}
+
+spec         | 
+-------------|------
+description  | Top-level map that includes secrets provider configuration [spec attributes][8].
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"spec": {
+  "client": {
+    "address": "https://vaultserver.example.com:8200",
+    "token": "VAULT_TOKEN",
+    "version": "v1",
+    "tls": {
+      "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem"
+    }
+  },
+  "max_retries": 2,
+  "timeout": "20s",
+  "rate_limiter": {
+    "limit": 10.0,
+    "burst": 100
+  }
+}
+{{< /highlight >}}
+
+### Metadata attributes
+
+name         |      |
+-------------|------
+description  | Provider name used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "vault"{{< /highlight >}}
+
+### Spec attributes
+
+client       | 
+-------------|------ 
+description  | Map that includes secrets provider configuration [client attributes][12].
+required     | true
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"client": {
+  "address": "https://vaultserver.example.com:8200",
+  "token": "VAULT_TOKEN",
+  "version": "v1",
+  "tls": {
+    "ca_cert": "/etc/ssl/certs/vault_ca_cert.pem"
+}
+{{< /highlight >}}
+
+max_retries  | 
+-------------|------ 
+description  | Number of times to retry connecting to the vault provider.
+required     | true
+type         | Integer
+default      | 2
+example      | {{< highlight shell >}}"max_retries": 2{{< /highlight >}}
+
+timeout      | 
+-------------|------ 
+description  | Provider connection timeout (hard stop).
+required     | false
+type         | String
+default      | 60s
+example      | {{< highlight shell >}}"timeout": "20s"{{< /highlight >}}
+
+rate_limiter | 
+-------------|------ 
+description  | Maximum rate and burst limits for the secrets API.
+required     | false
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"rate_limiter": {
+  "limit": 10.0,
+  "burst": 100
+}
+{{< /highlight >}}
+
+#### Client attributes
+
+address      | 
+-------------|------ 
+description  | Vault server address.
+required     | true
+type         | String
+example      | {{< highlight shell >}}
+"address": "https://vaultserver.example.com:8200"
+{{< /highlight >}}
+
+token        | 
+-------------|------ 
+description  | Vault token to use for authentication.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"token": "VAULT_TOKEN"{{< /highlight >}}
+
+version      | 
+-------------|------ 
+description  | HashiCorp Vault [version][14].
```

Comment:
The version refers to the API version, not the Vault version. The version can only be `v1` for production servers (see https://www.vaultproject.io/api-docs/#http-api) or `v2` for dev servers.